File: Actividad-2/servidor.js
import express from "express";

// Creamos la aplicación
const app = express();

// Puerto donde correrá el servidor
const PORT = 3000;

// Le decimos a Express que sirva archivos de la carpeta public
app.use(express.static("public"));

// Iniciar servidor
app.listen(PORT, () => {
  console.log(`Servidor funcionando en http://localhost:${PORT}`);
});

/* 
 Diferencia entre servidor.js y public/JS/index.js
servidor.js → Es el archivo que ejecuta Node.js para levantar el servidor web. 
Contiene la lógica del backend: configuración de Express, rutas, middlewares, etc. 
En tu caso, simplemente sirve archivos estáticos desde la carpeta public.

public/JS/index.js → Es un archivo JavaScript que se ejecuta en el navegador del cliente 
(frontend). No tiene nada que ver con Node.js directamente. 
Este archivo puede contener código para manipular el DOM, 
hacer peticiones al servidor, etc. Los usuarios lo 
descargan cuando visitan tu página.
*/

File: Actividad-3/servidor.py
import http.server
import socketserver

# Puerto donde correrá el servidor
PORT = 5000

# Configurar el manejador para servir archivos desde la carpeta "public"
Handler = http.server.SimpleHTTPRequestHandler
Handler.directory = "public"  # Carpeta raíz para los archivos estáticos

# Crear el servidor TCP
with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print(f"Servidor funcionando en http://localhost:{PORT}")
    # Mantener el servidor en ejecución hasta que se interrumpa (Ctrl+C)
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        print("\nServidor detenido.")